src/chunkserver/watchdog/watcher.h

```diff
@@ -0,0 +1,77 @@
+/*
+ *  Copyright (c) 2020 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Project: curve
+ * Created Date: 2020-12-11
+ * Author: qinyi
+ */
+
+#ifndef SRC_CHUNKSERVER_WATCHDOG_WATCHER_H_
+#define SRC_CHUNKSERVER_WATCHDOG_WATCHER_H_
+
+#include <memory>
+
+#include "src/chunkserver/watchdog/common.h"
+#include "src/chunkserver/watchdog/helper.h"
+
+namespace curve {
+namespace chunkserver {
+class Watcher {
+ public:
+    Watcher(volatile DogHealth* health, WatcherName watcherName,
```

Comment:
it's better to receive a dog in here rather then is internal state `health`, and check health through dog->IsHealth()

Reply:
done